### DISCUSSION TITLE
[Dev Tools] Hardhat zksync - Error verifying contract on abstract testnet

### DISCUSSION THREAD

**Discussion by coffee-the-dev** [zkSync-Community-Hub/zksync-developers#828] [Dev Tooling] [answered]
### Team or Project

Abstract

### Environment

Testnet

### Select the Dev Tool you are using

Hardhat Plugins

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Getting an error attempting to verify a contract I just deployed on abstract testnet:
```
Error in plugin @matterlabs/hardhat-zksync-verify: Backend verification error: Unknown zksolc version: v1.5.8

For more info run Hardhat with --show-stack-traces
```

**Comment by alhenaai** [comment]
It appears that the error you're encountering is due to the use of an unsupported version of `zksolc` in your Hardhat configuration. The `@matterlabs/hardhat-zksync-verify` plugin requires a specific version of `zksolc` that is compatible with the ZKsync network.

To resolve this issue, you should ensure that you are using a compatible version of `zksolc`. Typically, the latest version is recommended unless specified otherwise in the ZKsync documentation. You can update your `zksolc` version by modifying your Hardhat configuration file to match the supported version.

For more detailed guidance on setting up and configuring your Hardhat environment for ZKsync, including the correct versions of tools and plugins, please refer to the [ZKsync Hardhat plugin documentation](https://docs.zksync.io/zksync-era/tooling/hardhat/plugins/hardhat-zksync-verify).

If you continue to experience issues, I recommend visiting the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) for more in-depth support and troubleshooting.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment]
Hey @coffeexcoin 

I see the issue, let me see if I can quickly resolve as `v1.5.8` was recently released which should have been picked up by the verifier.

**Reply by dutterbutter** [reply]
The issue has been [resolved](https://github.com/matter-labs/zksync-era/pull/3378) and is part of the _next_ release [here](https://github.com/matter-labs/zksync-era/pull/3379). Apologies for the inconvenience here, the provided fix adds further automation which should prevent this in the future.

**Comment by shahar4** [comment, marked as answer]
@coffeexcoin @dutterbutter this is solved now. We've updated the verifier to automatically poll new compilers as @dutterbutter mentioned above, so this will not recur.
